### PR TITLE
support chunk partitioning for feature data

### PR DESCRIPTION
**Pull request by LiSu**
1. Add chunk partitioning for node feature
2. To support append write, use pkl to save feature file

### CODE REVIEW

File: graphlearn_torch/python/partition/base.py [hunk 1]
```diff
@@ -116,6 +116,58 @@ def save_feature_partition(
     torch.save(feature_partition.cache_feats, os.path.join(subdir, 'cache_feats.pt'))
     torch.save(feature_partition.cache_ids, os.path.join(subdir, 'cache_ids.pt'))
 
+def save_feature_partition(
```
Comment (baoleai) [inline graphlearn_torch/python/partition/base.py:119]: Now there are two duplicate methods?

File: graphlearn_torch/python/partition/base.py [hunk 2]
```diff
@@ -116,6 +116,58 @@ def save_feature_partition(
...
+  if graph_type is not None:
+    subdir = os.path.join(subdir, as_str(graph_type))
+  ensure_dir(subdir)
+  append_tensor_to_file(os.path.join(subdir, 'feats.pkl'), feature_partition.feats)
+  append_tensor_to_file(os.path.join(subdir,'ids.pkl'), feature_partition.ids)
+  if feature_partition.cache_feats is not None:
+    torch.save(feature_partition.cache_feats, os.path.join(subdir, 'cache_feats.pt'))
+    torch.save(feature_partition.cache_ids, os.path.join(subdir, 'cache_ids.pt'))
+
+def save_partition_feature_chunk(
```
Comment (baoleai) [inline graphlearn_torch/python/partition/base.py:138]: Should we use `save_feature_partition_chunk` to keep the same naming style?

File: graphlearn_torch/python/partition/base.py (hunk 1 above)
Comment (LiSu) [reply graphlearn_torch/python/partition/base.py:99]: removed the one that stores feature as pt file

File: graphlearn_torch/python/partition/base.py (hunk 2 above)
Comment (LiSu) [reply graphlearn_torch/python/partition/base.py]: renamed in the same style

**Review by baoleai — APPROVED** [review]

**Pull request merged**